reference-implementation/lib/abstract-ops/readable-streams.js

```diff
@@ -200,6 +200,12 @@ function ReadableStreamPipeTo(source, dest, preventClose, preventAbort, preventC
       }
 
       return transformPromiseWith(writer._readyPromise, () => {
+        if (shuttingDown === true) {
+          return promiseResolvedWith(true);
+        }
+        if (dest._state !== 'writable' || WritableStreamCloseQueuedOrInFlight(dest) === true) {
+          return promiseResolvedWith(true);
+        }
```

Comment:
This implements @domenic's suggestion from https://github.com/whatwg/streams/issues/1207#issuecomment-1015851512.

I don't know if we need to update the spec text for this. It already specifies that these checks *must* happen before performing *any* reads and writes:
> **Shutdown must stop activity:** if _shuttingDown_ becomes true, the user agent must not initiate further reads from _reader_, and must only perform writes of already-read chunks, as described below. In particular, the user agent must check the below conditions before performing any reads or writes, since they might lead to immediate shutdown.

We should still add a test for this particular case (although that might not be easy looking at the discussion in #1207).

Reply:
I personally don't think we need to update the spec text.